Homework3/LZW/LZW/LZWDecoder.cs

```diff
@@ -0,0 +1,121 @@
+namespace LZW;
+
+/// <summary>
+/// Class that represents LZW-decoder.
+/// </summary>
+public static partial class LZWDecoder
+{
+    /// <summary>
+    /// Decodes input array of LZW-encoded bytes.
+    /// </summary>
+    /// <param name="input">Input array of LZW-encoded bytes.</param>
+    /// <returns>Array of LZW-decoded bytes.</returns>
+    public static byte[] Decode(byte[]? input)
+    {
+        if (input == null)
+        {
+            throw new ArgumentNullException("Input bytes stream can't be null!");
+        }
+
+        if (input.Length == 0)
+        {
+            throw new ArgumentException("Input bytes stream can't be empty!");
+        }
+
+        var output = new List<byte>();
+        var dictionary = CreateAndInitDictionary();
```

Comment:
```suggestion
        Dictionary<int, List<byte>> dictionary = CreateAndInitDictionary();
```
А вот тут нужно тип выписать. Из названия метода неочевидно, какого именно типа результат.